analysis/src/SharedTypes.ml

```diff
@@ -433,6 +433,7 @@ module Completable = struct
     | Cpath of contextPath
```

Comment:
Would you start adding comments to Cpath and contextPath?
There's the question of whether there should be an entirely different case of type t, or whether typed context info should be integrated into the existing ones.

Reply:
Would you mind helping me out with adding comments to those that exist already? I have a hard time parsing what they are/mean.

Reply:
`Cpath` is for complex expressions

Reply:
`CPString` is for string constants, used in pipe expressions `34 -> ...`
`CPArray` is for arrays used in pipe expressions `[] -> ....`
 `CPApply of contextPath * Asttypes.arg_label list` is application with a list of labels
 
`CPId of string list * completionContext` is for a path `A.` or `A.B.C.` and the rhs is the rest to be completed

 `CPField of contextPath * string` is the `.z` of record access `expr.z` 
  `CPObj of contextPath * string` is the `["hello"]` of object access `expr["hello"]`
  `CPPipe of contextPath * string` is `expr->foo` where `foo` is the function whose first argument is `expr`.